src/vm/assembly.cpp

```diff
@@ -2588,6 +2588,7 @@ HRESULT RunMain(MethodDesc *pFD ,
             // Set the return value to 0 instead of returning random junk
             *pParam->piRetVal = 0;
             threadStart.Call(&stackVar);
+            *pParam->piRetVal = GetLatchedExitCode();
```

Comment:
@jkotas Thank you for comment. It seems that there is an issue in Legacy JIT. When I checked the JITed code via COMPlus_JitDisasm, it invokes Environment::set_ExitCode instead of Environment::Exit. I'll check that again.